docs_src/design/adr/core/0021-Device-Profile-Changes.md

```diff
@@ -0,0 +1,66 @@
+
+Please see a [prior PR](https://github.com/edgexfoundry/edgex-docs/pull/605) on this topic that detailed much of the debate and context on this issue.  For clarity and simplicity, that PR was closed in favor of this simpler ADR.
+
+## Context
+While the device profile has always been the way to describe a device/sensor and template its communications to the rest of the EdgeX platform, over the course of EdgeX evolution there have been changes in what could change in a profile (often based on its associations to other EdgeX objects).  This document is meant to address the issue of change surrounding device profiles in EdgeX going forward – specifically when can a device profile (or its sub-elements such as device resources) be added, modified or removed.
+
+### Summary of Device Profile Rules
+These rules will be implemented in core metadata on device profile API calls.
+
+- A device profile can be added anytime
+- Device resources or device commands can be added to a device profile anytime
+- Attributes can be added to a device profile anytime
+- A device profile can be removed or modified when the device profile is not associated to a device or provision watcher
+    - this includes modifying any field (except identifiers like names and ids)
+    - this includes changes to the array of device resources, device commands
+    - this includes changes to attributes (of device resources)
+    - even when a device profile is associated to a device or provision watcher, fields of the device profile or device resource can be modified when the field change will not affect the behavior of the system (example: the description field has no effect on the system).
+- A device profile cannot be removed when it is associated to a device or provision watcher.
+- A device profile can be removed or modified even when associated to an event or reading.  However, configuration options (see New Configuration Settings below) are available to block the change or removal of a device profile for any reason.
+    - the rationale behind the new configuraton settings was specifically to protect the event/reading association to device profiles.  Events and readings are generally considered short lived (ephemeral) objects and already contain the necessary device profile information that are needed by the system during their short life without having to refer to and keep the device profile.  But if an adopter wants to make sure the device profile is unmodified and still exists for any event/readings association (or for any reason), then the new config setting will block device profile changes or removals.
+    - see note below in Consequences that a new Units property must be added to the Reading object in order to support this rule and the need for all relevant profile data to be in the reading.
+
+### Ancillary Rules associated to Device Profiles
+- Identifying or “key” fields for device profiles, device resources, etc. cannot be modified and can never be null.
```

Comment:
Wouldn't it be better to say just simply say `Name` and `Id` fields in the device profile cannot be changed? Are there any other identifying or "key" fields?

Reply:
changed accordingly